### PR TITLE
Update narratives based on version in at narrative_version url

### PR DESCRIPTION
**Pull request by sychan**
This feature branch adds functionality to the reaper process that:

1) implements a query to a narrative_version endpoint ( currently configured in narrative_version_url defaulting to "https://narrative.kbase.us/narrative_version" ).
2) If the version query completes and returns a version string, it becomes the new required version (note - this can actually be a lower version than the previous value) and any existing prespawned narratives that have a lower version number get reaped.
3) Whether or not the required version is updated, the reaper checks to see if there are the required number of prespawned narratives - if not, enough new narratives are spawned to match cfg['num_prespawn']. This is how any old narratives reaped in step 2 above get replenished. This also allows admins to manually delete prespawned containers and have them replenished automatically.

### CODE REVIEW

**Comment by kkellerlbl** [comment]
"any existing prespawned narratives that have a lower version number get reaped."    Perhaps it is better to reap narratives with a different version, higher or lower?  This way we can roll back to a previous version of a narrative image if we need to.  Is there a downside to that?

File: app.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
        "kbase_cookie": u"kbase_session",
        "container_name": u"narrative-{}",
        "container_name_prespawn": u"narrativepre-{}",
+       "narrative_version_url": "https://narrative.kbase.us/narrative_version",
```
Comment (kkellerlbl) [inline app.py]: If we're setting defaults, I think they should all be for one deploy env.  So we should either change this to match using CI in auth2, or we should change the default value auth2 to use prod.  (I think I prefer defaulting to prod, but it's not a strong preference.)

File: app.py [hunk 2]
```diff
@@ -344,10 +346,65 @@ def get_active_traefik_svcs():
...
+            raise(Exception("Error querying {} for version: {} {}".format(cfg['narrative_version_url'],
+                            r.status_code, r.text)))
+    except Exception as err:
+        raise(err)
+    return(version)
+
+
+def reap_older_prespawn(version: str) -> None:
+    """
+    Reaps prespawned narratives that are older than the version string passed in
```
Comment (kkellerlbl) [inline app.py:383]: May be better to reap narratives with a different version instead of lower, to make rolling back easier?

**Review by kkellerlbl — CHANGES_REQUESTED** [review]
First pass.  I am still reading the new code in reaper() and in manage_rancher.

edit: reviewed other code, no new comments.

File: app.py (hunk 1 above)
Comment (sychan) [reply app.py]: Good point, I settled on ci for now.

File: app.py (hunk 2 above)
Comment (sychan) [reply app.py:383]: I've changed it to from less than to not-equal

**Comment by sychan** [comment]
@kkellerlbl Please look over the changes. I also added comments for all of the cfg entries, since I managed to forget a few of them after writing them 3 months ago!

**Review by bio-boris — APPROVED** [review]
Looks good to me.

**Comment by sychan** [comment]
@kkellerlbl @jsfillman Do you have an issues with me merging this this evening?

**Review by kkellerlbl — APPROVED** [review]
Looks good.

**Pull request merged**